src/executor.cpp

```diff
@@ -561,51 +551,51 @@ void Executor::execute01()
 
 void Executor::execute12()
 {
-    // Need to show blank spaces in case no srcData as function is enabled.
-    constexpr auto blankHexWord = "        ";
-    std::vector<std::string> output(4, blankHexWord);
+    displaySRCFunction(constants::FUNCTION_12);
+}
+
+void Executor::displaySRCFunction(const uint8_t function)
+{
+    // Need to show zeroes in case no srcData as function is enabled.
+    constexpr auto hexZeroes = "00000000";
+    std::vector<std::string> output(4, hexZeroes);
 
     if (!latestSrcAndHexwords.empty())
     {
-        std::vector<std::string> src;
-        boost::split(src, latestSrcAndHexwords, boost::is_any_of(" "));
+        auto sizeOfLatestSrcAndHexwords = latestSrcAndHexwords.size();
+        if (sizeOfLatestSrcAndHexwords < maxLengthOfSavedWords)
+        {
+            // appened zeroes at the end
```

Comment:
This should never happen , bail out here

Reply:
done